sematic/log_reader.py

```diff
@@ -0,0 +1,429 @@
+# Standard Library
+import base64
+import itertools
+import json
+from dataclasses import asdict, dataclass
+from typing import Iterable, List, Optional
+
+# Sematic
+from sematic import storage
+from sematic.abstract_future import FutureState
+from sematic.db.models.resolution import Resolution, ResolutionKind, ResolutionStatus
+from sematic.db.queries import get_resolution, get_run
+from sematic.resolvers.cloud_resolver import (
+    END_INLINE_RUN_INDICATOR,
+    START_INLINE_RUN_INDICATOR,
+)
+
+# Why the "V1"? Because we will likely want to change the structure of
+# the logs such that each file contains a different subset of logs. But
+# when we make this change, we will still want logs written in the old
+# structure to be readable, at least for a while. So we need to identify
+# which structure the files are in somehow, and a v1/v2 prefix is how we
+# can do it.
+V1_LOG_PATH_FORMAT = "logs/v1/run_id/{run_id}/{log_kind}/"
+
+
+def log_prefix(run_id: str, is_resolve: bool):
+    kind = "resolve" if is_resolve else "calculation"
+    return V1_LOG_PATH_FORMAT.format(run_id=run_id, log_kind=kind)
```

Comment:
Why not re-use the `JobType` enum from `sematic.scheduling.external_job`?
We can extract it in its own module to limit cross-dependencies if necessary. But in general, better to centralize naming and classifications.

Reply:
seemed a bit heavy when a bool would do, but sure.